src/main/java/net/starlark/java/eval/Starlark.java

```diff
@@ -244,7 +244,7 @@ public static void checkMutable(Mutability.Freezable x) throws EvalException {
   public static Object[] toArray(Object x) throws EvalException {
     // Specialize Sequence and Dict to avoid allocation and/or indirection.
     if (x instanceof Sequence) {
-      return ((Sequence<?>) x).toArray(new Object[((Sequence) x).size()]);
```

Comment:
This is not safe: Sequence may have arbitrary subclasses and the contract for List.toArray does not mandate that the resulting class is exactly Object[], yet that is a postcondition of Starlark.toArray and it is relied upon by StarlarkList (so that later setElement calls don't fail due to ArrayStoreException).

We should add a comment here explaining that.